File: app/src/main/java/com/kakacat/minitool/appInfo/model/AppDetailModel.java
package com.kakacat.minitool.appInfo.model;

import android.app.Activity;
import android.graphics.Bitmap;

import com.kakacat.minitool.appInfo.model.bean.AppInfoBean;
import com.kakacat.minitool.common.util.SystemUtil;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;

public class AppDetailModel {

    private static AppDetailModel model;

    private AppInfoBean appInfoBean;

    private AppDetailModel() {

    }

    public static AppDetailModel getInstance(){
        if(model == null){
            model = new AppDetailModel();
        }
        return model;
    }

    public AppInfoBean getAppInfoBean(Activity activity) {
        if (appInfoBean == null) {
            appInfoBean = activity.getIntent().getParcelableExtra("appInfo");
        }
        return appInfoBean;
    }

    public String saveIcon() {
        String result = null;
        try {
            String path = "/storage/emulated/0/MiniTool/" + appInfoBean.getAppName() + ".png";
            File file = new File(path);
            SystemUtil.createFile(file,false);

            FileOutputStream fos = new FileOutputStream(file);
            appInfoBean.getBitmap().compress(Bitmap.CompressFormat.PNG, 100, fos);
            fos.flush();
            fos.close();
            result = "成功保存在目录 : " + path;
        } catch (IOException e) {
            e.printStackTrace();
        }
        return result;
    }


    public void openMarket(Activity activity) {
        SystemUtil.openMarket(activity);
    }

    public String saveApk() {
        String srcPath = appInfoBean.getSourceDir();
        String desPath = "/storage/emulated/0/MiniTool/" + appInfoBean.getAppName() + ".apk";
        SystemUtil.createFile(new File(desPath),false);
        String[] commands = new String[]{
                "cp " + srcPath + " " + desPath + "\n",
        };
        SystemUtil.executeLinuxCommand(commands, false, true);
        return "提取成功 保存在" + desPath;
    }

    public String copyMd5(Activity activity) {
        SystemUtil.copyToClipboard(activity, "md5", appInfoBean.getSignMd5());
        return "复制成功";
    }

    public void openDetailInSetting(Activity activity) {
        SystemUtil.openAppDetailInSetting(activity,appInfoBean.getPackageName());
    }
}

File: app/src/main/java/com/kakacat/minitool/appInfo/model/AppDetailModel.kt
package com.kakacat.minitool.appInfo.model

import android.app.Activity
import android.graphics.Bitmap
import com.kakacat.minitool.appInfo.model.bean.AppInfoBean
import com.kakacat.minitool.common.util.SystemUtil
import java.io.File
import java.io.FileOutputStream

class AppDetailModel {

    private var appInfoBean: AppInfoBean?=null

    fun getAppInfoBean(activity: Activity): AppInfoBean? {
        when(appInfoBean){
            null -> appInfoBean = activity.intent.getParcelableExtra("appInfo")
        }
        return appInfoBean
    }

    fun saveIcon(): String {
        val result: String
        val path = "/storage/emulated/0/MiniTool/" + appInfoBean!!.appName + ".png"
        val file = File(path)
        SystemUtil.createFile(file,false)
        val fos = FileOutputStream(file)
        appInfoBean!!.bitmap!!.compress(Bitmap.CompressFormat.PNG,100,fos)
        fos.flush()
        fos.close()
        result = "成功保存在目录$path"
        return result
    }

    fun openMarket(activity: Activity){
        SystemUtil.openMarket(activity)
    }

    fun saveApk(): String{
        val srcPath = appInfoBean!!.sourceDir
        val desPath = "/storage/emulated/0/MiniTool/${appInfoBean!!.appName}.apk"
        SystemUtil.createFile(File(desPath),false)
        val commands = arrayOf("cp $srcPath $desPath \n")
        SystemUtil.executeLinuxCommand(commands,false,true)
        return "提取成功,保存在$desPath"
    }

    fun copyMd5(activity: Activity): String{
        SystemUtil.copyToClipboard(activity,"md5",appInfoBean!!.signMd5)
        return "复制成功"
    }

    fun openDetailInSetting(activity: Activity?) {
        SystemUtil.openAppDetailInSetting(activity!!, appInfoBean!!.packageName)
    }
}